dispatches/models/renewables_case/RE_flowsheet.py

```diff
@@ -0,0 +1,371 @@
+    if "detailed" in tank_type:
+        m.fs.h2_tank = HydrogenTank(default={"property_package": m.fs.h2ideal_props, "dynamic": False})
+        m.fs.h2_tank.tank_diameter.fix(0.1)
+        m.fs.h2_tank.tank_length.fix(length_m)
+        m.fs.h2_tank.control_volume.properties_in[0].pressure.setub(max_pressure_bar * 1e5)
+        m.fs.h2_tank.control_volume.properties_out[0].pressure.setub(max_pressure_bar * 1e5)
+        m.fs.h2_tank.previous_state[0].pressure.setub(max_pressure_bar * 1e5)
+        if tank_type == "detailed-valve":
+            # hydrogen tank valve
+            m.fs.tank_valve = Valve(
+                default={
+                    "valve_function_callback": ValveFunctionType.linear,
+                    "property_package": m.fs.h2ideal_props,
+                }
+            )
+            # connect tank to the valve
+            m.fs.tank_to_valve = Arc(
+                source=m.fs.h2_tank.outlet,
+                destination=m.fs.tank_valve.inlet
+            )
+            m.fs.tank_valve.outlet.pressure[0].fix(pem_pres_bar * 1e5)
+            # NS: tuning valve's coefficient of flow to match the condition
+            m.fs.tank_valve.Cv.fix(valve_cv)
+            # NS: unfixing valve opening. This allows for controlling both pressure
+            # and flow at the outlet of the valve
+            m.fs.tank_valve.valve_opening[0].unfix()
+            m.fs.tank_valve.valve_opening[0].setlb(0)
+    elif tank_type == "simple":
+        m.fs.h2_tank = SimpleHydrogenTank(default={"property_package": m.fs.h2ideal_props, "dynamic": False})
+        m.fs.h2_tank.outlet_to_turbine.mole_frac_comp[0, "hydrogen"].fix(1)
+        m.fs.h2_tank.outlet_to_pipeline.mole_frac_comp[0, "hydrogen"].fix(1)
+    else:
+        raise ValueError(f"Unrecognized tank_type {tank_type}")
+
+    m.fs.h2_tank.dt[0].fix(timestep_hrs * 3600)
+
+    return m.fs.h2_tank
+
+
+def add_h2_turbine(m, pem_pres_bar):
+    m.fs.h2turbine_props = GenericParameterBlock(default=hturbine_config)
+
+    m.fs.reaction_params = h2_reaction_props.H2ReactionParameterBlock(
+        default={"property_package": m.fs.h2turbine_props})
+
+    # Add translator block
+    m.fs.translator = Translator(
+        default={"inlet_property_package": m.fs.h2ideal_props,
+                 "outlet_property_package": m.fs.h2turbine_props})
+
+    m.fs.translator.eq_flow_hydrogen = Constraint(
+        expr=m.fs.translator.inlet.flow_mol[0] ==
+        m.fs.translator.outlet.flow_mol[0]
+    )
+
+    m.fs.translator.eq_temperature = Constraint(
+        expr=m.fs.translator.inlet.temperature[0] ==
+        m.fs.translator.outlet.temperature[0]
+    )
+
+    m.fs.translator.eq_pressure = Constraint(
+        expr=m.fs.translator.inlet.pressure[0] ==
+        m.fs.translator.outlet.pressure[0]
+    )
+
+    m.fs.translator.mole_frac_hydrogen = Constraint(
+        expr=m.fs.translator.outlet.mole_frac_comp[0, "hydrogen"] == 0.99
+    )
```

Comment:
Here, I would recommend using `fix` (`m.fs.translator.outlet.mole_frac_comp[0, "hydrogen"]`) instead of adding a constraint. This way, the variable for the mole fraction of hydrogen does not appear in the optimization model that the solver sees. While they are mathematically the same, sometimes the solver works better with the former.

Reply:
fixed